### PR TITLE
feat(mp-sjaz): rest-aware League match scheduling (no back-to-back)

### PR DESCRIPTION
**Pull request by gitrgoliveira**
## Summary

- League-format competitions no longer schedule a player to fight two matches in a row. Previously a full round-robin League was scheduled as a single Pool with independent per-court time cursors, so a player could be sent straight from one match into the next with no rest.
- League match scheduling now runs through a dedicated rest-aware scheduler that spreads every player's matches across time slots. Guarantees: no player is ever in two matches in the same slot (no cross-court double-booking), and no player is ever in two adjacent slots (no back-to-back), for any court count.
- Court count now only affects schedule length, not correctness: fewer courts means more idle rest slots. Small pools (Mixed format) and all other formats (Swiss, Playoffs) are unchanged, back-to-back in a 3-4 player pool is expected and stays.

## Why

In kendo, a small pool forces back-to-back play by pigeonhole, but a larger League should give players rest between matches. The old single-Pool model could not express that. A first design (a runtime "play rounds in order" gate) was rejected because it only enforced order, not rest. The final design makes no-back-to-back a hard scheduling invariant: whenever every remaining match would put a just-played player straight back in, the scheduler inserts an idle slot (which is the rest period) and advances. This always terminates (after an idle slot the just-played players are two slots clear, so a match always places next) and reuses the `Round`/circle-method infrastructure from mp-dc52 rather than duplicating it. The runtime simultaneity gate (`checkSimultaneousMatch`) is untouched and remains the defense-in-depth backstop at match start.

## Files changed

| File | What |
|---|---|
| `internal/engine/league_schedule.go` | New. `scheduleLeagueSlots` (greedy rest-aware slot assignment, hard no-back-to-back via idle-slot insertion) and `assignLeagueSlotTimes` (slot-band time assignment sharing a single global cursor). |
| `internal/engine/league_schedule_test.go` | New. Completeness + no-double-booking + no-back-to-back for n=2..16 at recommended and warning-zone court counts, single-court, a lunch-boundary case, and an integration test through `generatePools`. |
| `internal/engine/pools.go` | Route `CompFormatLeague` through the new scheduler; all other single-pool / Mixed paths unchanged (tournament load moved above the branch so both share it). |
| `internal/engine/scheduler_slots.go` | Extract shared `parseCeremonyParams` helper; dedupe the ceremony-block setup across the pool, bracket, and league slot assigners (DRY). |
| `internal/engine/league_allocation_test.go` | Update the obsolete "same-round matches on distinct courts" subtest to the new per-slot invariant, since matches of one round deliberately no longer map to one parallel time slot. |

## Test plan

- [x] `make go/test` passes (lint + security scan + tests) - green; `internal/engine` at 85.9% (above the 85% gate), gosec 0 issues, govulncheck clean, vitest 2030 pass
- [x] New/updated unit tests cover the change - the two guarantees are asserted by slot index for n=2..16 across court counts, plus a `generatePools` integration test and a lunch-boundary case
- [x] Manual browser verification (for `web-mobile/` or `web/` changes): N/A, backend-only change, no UI surface touched
- [x] Screenshots added above (REQUIRED for any UI-affecting change): N/A, no UI impact
- [x] Docs updated under `docs/` for any new or changed user-facing feature, flag, command, or behavior (`make docs/build` passes): N/A, no `docs/` page describes League match ordering; no new flag or command
- [x] No new console errors or warnings: N/A, backend-only

Closes mp-sjaz

### CODE REVIEW

File: internal/engine/league_schedule.go [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+// slots are the deliberate price of the rest guarantee.
+func scheduleLeagueSlots(matches []state.MatchResult, courts []string) (ordered []state.MatchResult, slots []int) {
+	if len(courts) == 0 {
+		courts = []string{""}
+	}
+	numCourts := len(courts)
+
+	// sentinel: a player not yet scheduled has lastSlot = math.MinInt/2 so
+	// (lastSlot[x] == slot-1) is false for any non-negative slot.
+	const absent = -(1 << 30)
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply internal/engine/league_schedule.go]: Fixed: reworded the comment to "an arbitrary large negative lastSlot" (the constant is `-(1<<30)`; `math.MinInt/2` was inaccurate). Class swept: no other stale `math.MinInt` references in the diff.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: internal/engine/league_allocation_test.go [hunk 2]
```diff
@@ -55,21 +55,23 @@ func tryStartLeague(t *testing.T, compID string, courts []string, players []stri
...
 		out[i] = fmt.Sprintf("P%d", i+1)
 	}
 	return out
 }
 
 // TestLeagueAllocation_EdgeCases pins how a league (single pool spanning the
 // whole roster) allocates its matches to courts across a range of roster/court
-// shapes. The core rules under test (engine/pools.go):
+// shapes. The core rules under test (engine/pools.go + engine/league_schedule.go):
 //   - a league spreads its matches across ALL assigned courts (no court idle),
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply internal/engine/league_allocation_test.go]: Reworded to your intent: "every court carries at least one match; individual time slots may still leave a court idle when the rest-aware scheduler inserts a rest band." Swept the sibling: the `pools.go` "so no court sits idle" comment and the "multi-court league uses every assigned court (no court idle)" subtest name were clarified the same way.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: internal/engine/league_schedule_test.go [hunk 3]
```diff
@@ -0,0 +1,391 @@
...
+				startMin := at.Hour()*60 + at.Minute()
+				byPlayer[m.SideA] = append(byPlayer[m.SideA], startMin)
+				byPlayer[m.SideB] = append(byPlayer[m.SideB], startMin)
+			}
+			for player, mins := range byPlayer {
+				sort.Ints(mins)
+				for i := 1; i < len(mins); i++ {
+					assert.GreaterOrEqualf(t, mins[i]-mins[i-1], 2*perMatch,
+						"G2 violation: player %q has back-to-back matches %d minutes apart", player, mins[i]-mins[i-1])
+				}
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply internal/engine/league_schedule_test.go]: Done exactly as suggested: the integration test now replays the same slot cursor (parseCeremonyParams + skipCeremonyBlocks + perMatch steps) to map each ScheduledAt back to its slot index, then asserts no player appears in consecutive slot indices. This detects a lunch-straddling slot-adjacency regression that the prior wall-clock minute-gap check could not. The pure-function `verifyG2` already checked by slot index; this brings the integration test in line.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: internal/engine/league_schedule_test.go [hunk 4]
```diff
@@ -0,0 +1,402 @@
...
+			verifyCompleteness(t, ordered, n)
+			verifyG1(t, ordered, slots)
+			verifyG2(t, ordered, slots)
+		})
+	}
+}
+
+// --- Warning zone: numCourts = floor(n/2) (one above SuggestedMaxCourts) ---
+// G2 is a hard invariant (idle slots are inserted when needed), so it holds
+// here too; the higher court count only means a longer schedule.
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply internal/engine/league_schedule_test.go]: Reworded: dropped the "only means a longer schedule" claim. The comment now states the test asserts correctness (G1/G2) not schedule length, and that court count changes how matches pack into slots but never whether the rest guarantee holds. (commit 061e63d8)

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
